software/contrib/cvecorder.py

```diff
@@ -0,0 +1,348 @@
+                self.clearCvrs('all')
+                #self.bankToSave = self.ActiveBank
+                #self.saveState()
+                # reverse the ActiveCvr increment caused by the initial button press
+                if self.ActiveCvr > 0:
+                    self.ActiveCvr -= 1
+                else:
+                    self.ActiveCvr = self.numCVR
+            # 500ms press clears the active bank
+            elif ticks_diff(ticks_ms(), b2.last_pressed()) >  500:
+                self.clearCvrs(self.ActiveBank)
+                self.bankToSave = self.ActiveBank
+                self.saveState()
+                # reverse the ActiveCvr increment caused by the initial button press
+                if self.ActiveCvr > 0:
+                    self.ActiveCvr -= 1
+                else:
+                    self.ActiveCvr = self.numCVR
+
+        # B2 short press
+        @b2.handler
+        def b2Pressed():
+            # Change the active recorder channel
+            if self.ActiveCvr < self.numCVR:
+                self.ActiveCvr += 1
+            else:
+                self.ActiveCvr = 0
+
+    def handleClock(self):
+
+        # Sample input to 2 decimal places
+        self.CvIn = round(20 * ain.percent(), 2)
+
+        # Start recording if pending and on first step
+        if self.step == 0 and self.CvRecording[self.ActiveCvr] == 'pending':
+            self.CvRecording[self.ActiveCvr] = 'true'
+
+        for i in range(self.numCVR+1):
+            # If recording, write the sampled value to the CVR list and play the voltage
+            if self.CvRecording[i] == 'true':
+                self.CVR[self.ActiveBank][self.ActiveCvr][self.step] = self.CvIn
+                cvs[self.ActiveCvr].voltage(self.CvIn)
+            else:
+                cvs[i].voltage(self.CVR[self.ActiveBank][i][self.step])
+
+        # Reset step number at stepLength -1 as pattern arrays are zero-based
+        if self.step < self.stepLength - 1:
+            self.step += 1
+        else:
+            # Reset step to zero , stop recording and save recording to local storage
+            self.step = 0
+            if self.CvRecording[self.ActiveCvr] == 'true':
+                self.CvRecording[self.ActiveCvr] = 'false'
+                self.bankToSave = self.ActiveBank
+                self.saveState()
+
+    def clearCvrs(self, bank):
+        for b in range(self.numCVRBanks+1):
+            # skip bank unless 'all' is passed
+            if b != bank and bank != 'all':
+                continue
+            if self.debugTest:
+                print('Clearing bank: ' + str(b))
+            # Set all CV values to zero
+            for i in range(self.numCVR+1):
+                for n in range (0, self.stepLength):
+                    self.CVR[b][i][n] = 0
+            # Save the cleared bank to local storage
+            self.bankToSave = b
+            self.saveState()
+
+    def initCvrs(self):
```

Comment:
I would like to keep the initCvrs function in case I use it again in future versions. Please let me know how you would like me to comment this so you can ignore it in your PR review, thanks.